### PR TITLE
[TopBar] adopt search results to match width and alignment of input field

### PR DESCRIPTION
**Pull request by eliias**
### WHY are these changes introduced?

This addresses some of the discussions that we had after this issue popped up https://github.com/Shopify/polaris-react/issues/956 and is also a follow up of https://github.com/Shopify/polaris-react/pull/852. The original PR did change the width of the search input field in the TopBar but did not apply those changes to the search results container. This looks a bit weird and this PR aligns the search result with the input field and also matches the width of the field.

### WHAT is this pull request doing?

It sets the position rule to be relative on the search container. This allows us to match the width with the absolutely positioned search results container.

screen shot 2019-02-04 at 4 50 09 pm

screen shot 2019-02-04 at 4 57 10 pm

## 

### How to 🎩

🖥 [Local development instructions](https://github.com/Shopify/polaris-react/blob/master/README.md#development)
🗒 [General tophatting guidelines](https://github.com/Shopify/polaris-react/blob/master/documentation/Tophatting.md)
📄 [Changelog guidelines](https://github.com/Shopify/polaris-react/blob/master/documentation/Versioning%20and%20changelog.md)

<details>
<summary>Copy-paste this code in <code>playground/Playground.tsx</code>:</summary>

```jsx
import * as React from 'react';
import {ActionList, AppProvider, Card, Frame, Page, TopBar} from '../src';

interface State {}

export default class Playground extends React.Component<{}, State> {
  state = {
    userMenuOpen: false,
    searchActive: false,
    searchText: '',
  };

  toggleUserMenu = () => {
    // @ts-ignore
    this.setState(({userMenuOpen}) => ({userMenuOpen: !userMenuOpen}));
  };

  handleSearchResultsDismiss = () => {
    this.setState(() => {
      return {
        searchActive: false,
        searchText: '',
      };
    });
  };

  handleSearchChange = (value) => {
    this.setState({searchText: value});
    if (value.length > 0) {
      this.setState({searchActive: true});
    } else {
      this.setState({searchActive: false});
    }
  };

  render() {
    const {
      state,
      handleSearchChange,
      handleSearchResultsDismiss,
      toggleUserMenu,
    } = this;

    const {userMenuOpen, searchText, searchActive} = state;

    const theme = {
      colors: {
        topBar: {
          background: '#357997',
        },
      },
      logo: {
        width: 124,
        topBarSource:
          'https://cdn.shopify.com/s/files/1/0446/6937/files/jaded-pixel-logo-color.svg?6215648040070010999',
        url: 'http://jadedpixel.com',
        accessibilityLabel: 'Jaded Pixel',
      },
    };

    const userMenuMarkup = (
      <TopBar.UserMenu
        actions={[
          {
            items: [{content: 'Back to Shopify', icon: 'arrowLeft'}],
          },
          {
            items: [{content: 'Community forums'}],
          },
        ]}
        name="Dharma"
        detail="Jaded Pixel"
        initials="D"
        open={userMenuOpen}
        onToggle={toggleUserMenu}
      />
    );

    const searchResultsMarkup = (
      <Card>
        <ActionList
          items={[
            {content: 'Shopify help center'},
            {content: 'Community forums'},
          ]}
        />
      </Card>
    );

    const searchFieldMarkup = (
      <TopBar.SearchField
        onChange={handleSearchChange}
        value={searchText}
        placeholder="Search"
      />
    );

    const topBarMarkup = (
      <TopBar
        showNavigationToggle={true}
        userMenu={userMenuMarkup}
        searchResultsVisible={searchActive}
        searchField={searchFieldMarkup}
        searchResults={searchResultsMarkup}
        onSearchResultsDismiss={handleSearchResultsDismiss}
        onNavigationToggle={() => {
          console.log('toggle navigation visibility');
        }}
      />
    );

    return (
      <Page>
        <AppProvider theme={theme}>
          <Frame topBar={topBarMarkup} />
        </AppProvider>
      </Page>
    );
  }
}
```

</details>

### 🎩 checklist

* [x] Tested on [mobile](https://github.com/Shopify/polaris-react/blob/master/documentation/Tophatting.md#cross-browser-testing)
* [x] Tested on [multiple browsers](https://help.shopify.com/en/manual/intro-to-shopify/shopify-admin/supported-browsers)

### CODE REVIEW

File: src/components/TopBar/components/Search/Search.scss [hunk 1]
```diff
@@ -9,21 +9,27 @@ $search-results-desktop-large-width-offset: layout-width(secondary, min) +
...
-    left: 0;
+  @include breakpoint-after($not-condensed-content) {
+    padding: 0 spacing(loose);
+    width: 100%;
+    justify-content: left;
   }
 
-  @include page-content-when-not-fully-condensed {
-    align-items: flex-start;
+  @include breakpoint-after(743px) {
```
Comment (kvendrik) [inline src/components/TopBar/components/Search/Search.scss]: `743px` maybe store this as a variable somewhere

File: src/components/TopBar/components/Search/Search.scss [hunk 2]
```diff
@@ -9,21 +9,27 @@ $search-results-desktop-large-width-offset: layout-width(secondary, min) +
 .Search {
   position: absolute;
   z-index: z-index(search, $stacking-order);
-  top: top-bar-height();
-  left: $search-results-mobile-nav-toggle-offset;
-  display: flex;
-  align-items: stretch;
-  width: 100vw;
-  height: calc(100vh - #{top-bar-height() + spacing(tight)});
+  top: top-bar-height() - spacing();
```
Comment (kvendrik) [inline src/components/TopBar/components/Search/Search.scss]: Might wanna store this (or just the `spacing()` part) in a variable to create clarity around how this calculation works.

**Comment by eliias** [comment]
There was still a small issue w/ alignment on mobile. So we took the chance to also reposition the results a bit. Thx for review/polish @kvendrik.

screen shot 2019-02-05 at 6 11 13 pm

File: src/components/TopBar/components/Search/Search.scss [hunk 3]
```diff
@@ -33,37 +38,30 @@ $search-results-desktop-large-width-offset: layout-width(secondary, min) +
...
   width: 100%;
-
-  @include frame-when-nav-displayed {
-    width: calc(100% - #{$search-results-desktop-width-offset};);
-  }
-
-  @include breakpoint-after($large-width) {
-    width: calc(100% - #{$search-results-desktop-large-width-offset});
-  }
+  max-width: 70rem;
```
Comment (kvendrik) [inline src/components/TopBar/components/Search/Search.scss]: Maybe store these numbers somewhere to make them a little less *magical*.

Comment (eliias) [reply src/components/TopBar/components/Search/Search.scss]: I have extracted both values into variables and got them hoisted to the top of the file.

**Comment by eliias** [comment]
Hey folks, we want to get this shipped w/ the next release of Polaris. Please provide your final feedback or please give it a thumbs up. @jakob-stecher @ry5n.

**Pull request merged**